lua/possession/config.lua

```diff
@@ -150,6 +150,11 @@ local function fix_compatibility(opts)
             enable = opts.telescope.previewer,
         }
     end
+
+    local autoload = vim.tbl_get(opts, 'autoload')
+    if type(autoload) == 'table' then
+        opts.autoload = autoload.cwd and 'auto_cwd' or false
+    end
```

Comment:
This doesn't actually seem to work, because `warn_on_unknown_keys` executes before `fix_compatibility` in `setup` in this file.

If I comment out `warn_on_unknown_keys` then this backward compatibility feature looks to work.

The new autoload choices is a breaking change. Let me know how you want to handle it. The code looks like you supported backwards compatibility at some stage.

Reply:
Hmm, it looks like `fix_compatibility` should come before `warn_on_unknown_keys`. If you swap the order it should be fine?

Reply:
Looks like swapping the two function calls works.

Reply:
I've added a deprecated config warning, what do you think of this idea? I do note that for me at least, the warning flashes and disappears on load, replaced by the 'loaded session...' text. So users might not notice it. If you like this idea and have improvements for it, let me know. I can also undo it if you feel there is no benefit in it.

Reply:
I think that deprecation warning in this case will be nice, since we're simplifying/reducing this config option (from `autoload = { cwd = true }` to `autoload = '...'`. But for options that are becoming more complex I think deprecations are unnecessary (e.g. `feature = true` to `feature = { enabled = true, foo = 'bar' }`).

Reply:
Changed to `vim.deprecate`